src/main/java/org/llorllale/cactoos/matchers/Assertion.java

```diff
@@ -0,0 +1,143 @@
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.llorllale.cactoos.matchers;
+
+import org.cactoos.Scalar;
+import org.cactoos.scalar.UncheckedScalar;
+import org.hamcrest.Description;
+import org.hamcrest.Matcher;
+import org.hamcrest.StringDescription;
+
+/**
+ * An assertion that can be affirmed in tests.
+ * <p>
+ * Assertions are always performed on <em>operations</em>, as opposed to actual
+ * values, in order to also enable asserting on errors thrown by the former.
+ * <p>
+ * <strong>Examples:</strong>
+ * <p>
+ * Testing a value:
+ * <pre>
+ * {@code
+ *     public void test() {
+ *         new Assertion(
+ *             "must match the string",
+ *             () -> new TextOf("string"),
+ *             new TextIs("string")
+ *         ).affirm();    // value is affirmed
+ *     }
+ * }
+ * </pre>
+ *
+ * Testing an error (see {@link Throws}):
+ * <pre>
+ * {@code
+ *     public void test() {
+ *         new Assertion(
+ *             "must match the string",
+ *             () -> { throw new IllegalArgumentException("error msg"); },
+ *             new Throws("error msg", IllegalArgumentException.class)
+ *         ).affirm();    // error is affirmed
+ *     }
+ * }
+ * </pre>
+ *
+ * @param <T> The type of the result returned by the operation under test
+ * @since 1.0.0
+ * @todo #18:30min Replace all uses of MatcherAssert.assertThat() with
+ *  Assertion, and ban all overloads of the former in forbidden-apis.txt.
+ *  We should also look into banning common matchers like Matchers.is(), etc.
+ * @todo #18:30min Assertion relies on the operation under test to be idempotent
+ *  in order to match "error matchers" such as `Throws` as expected. This may
+ *  not be the case for all operations and could be a problem. Investigate if
+ *  there's a way around/through this.
+ */
+@SuppressWarnings("PMD.AvoidCatchingGenericException")
+public final class Assertion<T> {
+    /**
+     * Reason for refuting this assertion.
+     */
+    private final String reason;
+    /**
+     * Behaviour to test.
+     */
+    private final Scalar<T> test;
+    /**
+     * Matcher against which the behaviour will be tested.
+     */
+    private final Matcher<T> matcher;
+
+    /**
+     * Ctor.
+     * @param reason Reason for refuting this assertion
+     * @param test The behaviour to test
+     * @param matcher Matcher to test behaviour
+     */
+    public Assertion(
+        final String reason, final Scalar<T> test, final Matcher<T> matcher
```

Comment:
@llorllale, can we use the following formatting for the methods and ctors always?

Reply:
@dgroup totally agree

Reply:
@dgroup well, not *always* - only when the line length limit is exceeded (obviously)

Reply:
Yes, for that's purpose only.